hlapipeline/alignimages.py

```diff
@@ -175,127 +176,190 @@ def perform_align(input_list, archive=False, clobber=False, update_hdr_wcs=False
+        input image.  This list should have been  updated, in-place, with the new RMS values;
+        specifically,
+
+            * 'FIT_RMS': RMS of the separations between fitted image positions and reference positions
+            * 'TOTAL_RMS': mean of the FIT_RMS values for all observations
+            * 'NUM_FITS': number of images/group_id's with successful fits included in the TOTAL_RMS
+
+        These entries are added to the 'tweakwcs_info' dictionary.
+
+    Returns
+    -------
+    max_rms_val : float
+        The best Totol rms dteremined from all of the images
+
+    num_xmatches: int
+        The number of stars used in matching the data
+
+    """
+    tweakwcs_info_keys = OrderedDict(imglist[0].meta['tweakwcs_info']).keys()
+    imgctr=0
+    max_rms_val = 1e9
+    num_xmatches = 0
+    for item in imglist:
+        image_name = item.meta['name']
+        #Handle fitting failures (no matches found)
+        if item.meta['tweakwcs_info']['status'].startswith("FAILED") == True:
+                print("No cross matches found in any catalog for {} - no processing done.".format(image_name))
+                continue
+        fit_rms_val = item.meta['tweakwcs_info']['FIT_RMS']
+        max_rms_val = item.meta['tweakwcs_info']['TOTAL_RMS']
+        num_xmatches = item.meta['tweakwcs_info']['nmatches']
+        if num_xmatches < MIN_CROSS_MATCHES:
+            if catalogIndex < numCatalogs-1:
+                print("Not enough cross matches found between astrometric catalog and sources found in {}".format())
+                continue
+        print('RESULTS FOR {} Chip {}: FIT_RMS = {} mas, TOTOAL_RMS = {} mas, NUM =  {}'.format(image_name, item.meta['chip'], fit_rms_val, max_rms_val, num_xmatches))
+        # print fit params to screen
+        if print_fit_parameters:
+            print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ FIT PARAMETERS ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+            print("image: {}".format(image_name))
+            print("chip: {}".format(item.meta['chip']))
+            print("group_id: {}".format(item.meta['group_id']))
+            for tweakwcs_info_key in tweakwcs_info_keys:
+                if not tweakwcs_info_key.startswith("matched"):
+                    print("{} : {}".format(tweakwcs_info_key,item.meta['tweakwcs_info'][tweakwcs_info_key]))
+            # print("Radial shift: {}".format(math.sqrt(item.meta['tweakwcs_info']['shift'][0]**2+item.meta['tweakwcs_info']['shift'][1]**2)))
+            print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+
+    if max_rms_val > MAX_FIT_RMS:
+        print("Total fit RMS value = {} mas greater than the maximum threshold value {}.".format(max_rms_val, MAX_FIT_RMS))
+        print("Try again with the next catalog")
+    else:
+        print("Fit calculations successful.")
+
+    return max_rms_val, num_xmatches
```

Comment:
The current logic, as it appears to me, has this function returning the RMS value and number of cross-matches for the last group_id only, as opposed to any VISIT level value.